### PR TITLE
Shadow Demon Sound Rework

### PR DESCRIPTION
**Pull request by Stoniest**
## What Does This PR Do

Adds new unique sounds specifically for the Shadow Demon.

## Why It's Good For The Game

Makes the Shadow demon sound more menacing and adds to their theme of darkness and primal fears.

## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/115121249/90535430-8d10-4e07-95a0-0ee4f28afb2d

https://github.com/ParadiseSS13/Paradise/assets/115121249/f71ac4f8-ae40-4cd0-8340-ee97fa1f9efc

https://github.com/ParadiseSS13/Paradise/assets/115121249/385a589f-83e6-48fd-891d-ba009e066aaf

https://github.com/ParadiseSS13/Paradise/assets/115121249/6249fbc4-259a-4161-8bda-d401a049bf62

## Testing

Loaded up
Spawn humanoids and shadow demons
Smash stuff
Made cocoons
Grappled many things
Turn on hallucinations on cocoons
Let the shadow demon die many times
Hit more stuff

## Changelog
:cl:
soundadd: Added unique sounds to the shadow demons attacks, death, hallucinations, grappling and cocooning.
tweak: Tweaked some of the hallcuination sounds to be unique by switching some not so fitting/ignorable sounds.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm [hunk 1]
```diff
@@ -7,6 +7,7 @@
 	move_resist = MOVE_FORCE_STRONG
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE // so they can tell where the darkness is
 	loot = list(/obj/item/organ/internal/heart/demon/shadow)
+	death_sound = 'sound/shadowdemon/shadowdeath.ogg'///Adds death sound unique to shadow demon
```
Comment (Contrabang) [inline code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm:10]:
```suggestion
	death_sound = 'sound/shadowdemon/shadowdeath.ogg'
```

File: code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm [hunk 2]
```diff
@@ -35,6 +36,7 @@
 		adjustBruteLoss(-20)
 
 /mob/living/simple_animal/demon/shadow/UnarmedAttack(atom/A)
+	attack_sound = pick('sound/shadowdemon/shadowattack2.ogg', 'sound/shadowdemon/shadowattack3.ogg', 'sound/shadowdemon/shadowattack4.ogg')///Adds attack sounds unique to shadow demon
```
Comment (Contrabang) [inline code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm:39]:
```suggestion
	// Pick a random attack sound for each attack
	attack_sound = pick('sound/shadowdemon/shadowattack2.ogg', 'sound/shadowdemon/shadowattack3.ogg', 'sound/shadowdemon/shadowattack4.ogg')
```

File: code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm [hunk 3]
```diff
@@ -79,6 +81,7 @@
 
 /obj/structure/shadowcocoon/Initialize(mapload)
 	. = ..()
+	playsound(loc, 'sound/shadowdemon/shadownode.ogg', 5, TRUE, -1) ///plays sound when finished wrapping a body
```
Comment (Contrabang) [inline code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm:84]:
```suggestion
	playsound(loc, 'sound/shadowdemon/shadownode.ogg', 5, TRUE, extra_range = -1) ///plays sound when finished wrapping a body
```

Comment (Contrabang) [inline code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm:84]:
```suggestion
	playsound(loc, 'sound/shadowdemon/shadownode.ogg', 5, TRUE, -1)
```

File: code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm [hunk 4]
```diff
@@ -107,10 +110,12 @@
 			flare_to_darken.visible_message("<span class='notice'>[flare_to_darken] suddenly dims.</span>")
 		to_darken.extinguish_light()
 	if(!silent && time_since_last_hallucination >= rand(8, 12))
-		playsound(src, pick('sound/items/deconstruct.ogg', 'sound/weapons/handcuffs.ogg', 'sound/machines/airlock_open.ogg',  'sound/machines/airlock_close.ogg', 'sound/machines/boltsup.ogg', 'sound/effects/eleczap.ogg', get_sfx("bodyfall"), get_sfx("gunshot"), 'sound/weapons/egloves.ogg'), 50)
+		playsound(src, pick('sound/shadowdemon/shadowhalluc1.ogg', 'sound/shadowdemon/shadowhalluc2.ogg', 'sound/machines/airlock_open.ogg',  'sound/machines/airlock_close.ogg', 'sound/machines/boltsup.ogg', 'sound/shadowdemon/shadowhalluc3.ogg', get_sfx("bodyfall"), 'sound/weapons/egloves.ogg'), 50)
 		time_since_last_hallucination = 0
+///custom sounds added to hallucination to be more creepy and unnerving
```
Comment (Contrabang) [inline code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm:115]:
```suggestion
```

File: code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm [hunk 5]
```diff
@@ -107,10 +110,12 @@
...
 
-/obj/structure/shadowcocoon/AltClick(mob/user)
+
+/obj/structure/shadowcocoon/AltClick(mob/user)///Allows you to turn on cocoons making hallucination sounds or not
```
Comment (Contrabang) [inline code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm:118]:
```suggestion
// Allows you to turn on cocoons making hallucination sounds or not
/obj/structure/shadowcocoon/AltClick(mob/user)
```

File: code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm [hunk 6]
```diff
@@ -194,6 +199,7 @@
 	name = "shadow hand"
 	icon_state = "shadow_hand"
 	plane = FLOOR_PLANE
+	hitsound = 'sound/shadowdemon/shadowattack1.ogg'///Plays sound when grabbing something living or light
```
Comment (Contrabang) [inline code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm:202]:
```suggestion
	hitsound = 'sound/shadowdemon/shadowattack1.ogg' // Plays when hitting something living or a light
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
Some small stuff, we dont really want comments on every line, plus some comment standarization/better explanations

**Review by S34NW — APPROVED** [review]

**Pull request merged**